### DISCUSSION TITLE
Comment list is always empty

### DISCUSSION THREAD

**Discussion by Feuerfuchs** [posativ/isso#782] [Q&A]
I just switched from the last stable version to the master branch of this repository. Now the comment list API endpoint (`/?uri=...`) is broken for me since it will always return a 404 error, regardless of how many comments there actually are. The count API endpoint (`/count?uri=...`) works fine, though.

~~You can see this problem live here: https://newtest.feuerfuchs.eu/en/blog/2014/05/02/another-hugo-post/~~ (Comments are now disabled, URLs below still work)
Fetch API result: https://comments.feuerfuchs.eu/?uri=/en/blog/2014/05/02/another-hugo-post/
Count API result: https://comments.feuerfuchs.eu/count?uri=/en/blog/2014/05/02/another-hugo-post/
[Admin interface screenshot](https://user-images.githubusercontent.com/14197353/40931270-bede5a7c-682a-11e8-80b8-b3baab66064a.png)

**Comment by pozitron57** [comment]
Do you have the same problem when there are more than zero comments? See #301, isso returns 404 when there are no comments.

**Comment by Feuerfuchs** [comment]
Yes, that's the point of my issue. On my website, you will see in the article header that there are currently 2 comments for this article (which is the number reported by `/count?uri=...`). However, if you scroll down to the actual comments list, it says "No Comments Yet" (as reported by `/?uri=...`).

The comments must show up here because I posted them myself on that page and the admin panel shows a matching URI for the comments (see the screenshot I linked). That means it's not an issue with the count API endpoint but the fetch API endpoint.

**Comment by pozitron57** [comment]
Now I see, sorry, didn't read your post attentively enough.

**Comment by Feuerfuchs** [comment]
No problem, my description wasn't quite clear about what the issue actually is. I fixed it now. :)

**Comment by jelmer** [comment]
can you still reproduce this with 0.11.1?

**Comment by Feuerfuchs** [comment]
I will check as soon as possible! It might take a few days, though, since I'm very busy right now.

**Comment by Feuerfuchs** [comment]
That went faster than expected so here we go: The issue is still there, unfortunately.
The comment list returns error 404: https://comments.feuerfuchs.eu/?uri=/en/blog/2014/05/02/another-hugo-post/
The counter reports 2: https://comments.feuerfuchs.eu/count?uri=/en/blog/2014/05/02/another-hugo-post/
The admin interface lists 2 comments:
[screenshot_2018-11-26 isso admin](https://user-images.githubusercontent.com/14197353/49016102-5becf380-f185-11e8-9635-5a2e8e7826bc.png)
Isso version is 0.11.1:
![bildschirmfoto vom 2018-11-26 14-13-48](https://user-images.githubusercontent.com/14197353/49016152-7aeb8580-f185-11e8-9e19-b93a383e0e5a.png)

edit: Also, here's the URL where the comments are supposed to be displayed: https://www.feuerfuchs.eu/en/blog/2014/05/02/another-hugo-post/
As you can see, under the subtitle it says "2 Comments", but the actual comment list says "No Comments Yet".

**Comment by ix5** [comment]
Is this still current?

**Comment by fahmifan** [comment]
I still get empty comments with `isso 0.12.5` running in local.
The data exists in DB, the query seems to working too

![image](https://user-images.githubusercontent.com/19581621/155507424-61091dc3-066b-480a-a6c4-cfd1e0b9159b.png)

**Reply by ix5** [reply]
Could you please post something that makes it possible to reproduce your issue? A screenshot is of little to no value when verifying.

At the very least, your isso config, server config, embed code on the site and, if possible, a sanitized DB dump are needed.

**Reply by fahmifan** [reply]
Sorry, I found the issue, apparently, it was the config, I set the moderation to `true`, but I expect it to return the result immediately

**Reply by ix5** [reply]
What do you mean, "return the result immediately"? What is the expected behaviour?

**Reply by ix5** [reply]
Also note, the original poster had already approved the comments (or had no moderation enabled) and they still did not show up, so I do not think your moderation-related issue is the same as OP's.

**Reply by fahmifan** [reply]
Yeah, I think my issue is not related to the OP's. 
Actually, I am not sure what the cause of my issue is, but after I turn the moderation off in config, the comment showed up after I create it and refresh the page.

**Comment by ycpei** [comment]
I had the same problem, my config uses an nginx reverse proxy with sub-uri and https redirection. My location and X-Script-Name fields are different, and by making them the same, I got into a 308 redirect loop.  Following hints at https://github.com/posativ/isso/issues/622#issuecomment-1030901659, I reverted to my original `location` but added a trailing slash which fixed it.
